helix-vcs/src/git/test.rs

```diff
@@ -0,0 +1,106 @@
+use std::{fs::File, io::Write, path::Path, process::Command};
+
+use tempfile::TempDir;
+
+use crate::{DiffProvider, Git};
+
+fn exec_git_cmd(args: &str, git_dir: &Path) {
```

Comment:
I recommend trying `git-testtools` here as it allows to write [fixture as a script](https://github.com/Byron/gitoxide/blob/13356184c735d72edb891d64b3de1bb5c981a6ad/git-refspec/tests/fixtures/match_baseline.sh#L18) which can be invoked [like that](https://github.com/Byron/gitoxide/blob/429cccc5831c25a7205a12dc7a0443ac48616e2c/git-refspec/tests/matching/mod.rs#L212). It takes care of isolating `git` invocations for you as well. These invocations will naturally be cached as well.

Maybe it's too much here but if this test-suite may grow, it might be worth formalizing fixture generation more.

Reply:
For now the testsuite is simple enough that I prefer our current approach because it keeps compiletimes lower.
That being said git-testtools is a really great tool so I will definitely consider switching if the helix git testsuite grows beyond a couple simple tests, thanks for recommending it.
For now I copied the environment variable used for config isolation in git-testtools into the helix testsuite